libmachine/persist/kvstore.go

```diff
@@ -0,0 +1,145 @@
+package persist
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/docker/machine/libmachine/host"
+	"github.com/docker/machine/libmachine/kv"
+	"github.com/docker/machine/libmachine/log"
+)
+
+func init() {
+	// XXX do we need this?
+	//etcd.Register()
+	//consul.Register()
+	//zookeeper.Register()
+	//boltdb.Register()
+}
+
+type Kvstore struct {
+	Path string // compat
+}
+
+func NewKvstore(path string) *Kvstore {
+	log.Debugf("AK: NewKvstore(%s)", path)
+	kvurl, err := url.Parse(path)
+	if err != nil {
+		panic(fmt.Sprintf("Malformed store path: %s %s", path, err))
+	}
+	switch kvurl.Scheme {
```

Comment:
I think this should move into kv.Connect so this is a thinner wrapper, and the kv abstraction layer deals with the specific kv store stuff.